### PR TITLE
Fix SPIR-V friendly IR for OpGenericCastToPtr instruction

### PR DESCRIPTION
**Pull request by Fznamznon**
This instruction was translated directly to OCL even if SPIR-V
friendly LLVM IR was requested.

### CODE REVIEW

**Comment by Fznamznon** [comment]
@AlexeySotkin , @AlexeySachkov , @MrSidims please take a look.

File: lib/SPIRV/SPIRVToOCL.cpp [hunk 1]
```diff
@@ -515,6 +519,32 @@ void SPIRVToOCLBase::visitCallSPIRVImageMediaBlockBuiltin(CallInst *CI, Op OC) {
...
+        switch (AddrSpace) {
+        case SPIRAS_Global:
+          return std::string(kOCLBuiltinName::ToGlobal);
+        case SPIRAS_Local:
+          return std::string(kOCLBuiltinName::ToLocal);
+        case SPIRAS_Private:
+          return std::string(kOCLBuiltinName::ToPrivate);
+        default:
+          llvm_unreachable("Invalid address space");
+          return std::string();
```
Comment (MrSidims) [inline lib/SPIRV/SPIRVToOCL.cpp:542]: To fulfill my curiosity: compiler emits a warning if we don't add a return here?

**Review by MrSidims — APPROVED** [review]

File: lib/SPIRV/SPIRVToOCL.cpp (hunk 1 above)
Comment (Fznamznon) [reply lib/SPIRV/SPIRVToOCL.cpp:542]: Yes.

**Comment by Fznamznon** [comment]
Ping!

**Comment by Fznamznon** [comment]
I guess test/OpSpecConstant.spvasm should be fixed by https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1031

**Review by AlexeySotkin — APPROVED** [review]

**Pull request merged**